app/models/product.rb

```diff
@@ -52,6 +52,13 @@ def self.exclude(exclusion_list)
     where("products.id NOT IN (?)", exclusion_list)
   end
 
+  scope :for_facility, lambda { |facility|
+    return self.class.none if facility.blank?
+    return where(facility_id: facility.id) if facility.single_facility?
+    return all if facility.cross_facility?
+    self.class.none
```

Comment:
It should never hit it, but it felt wrong to have a scope potentially return nothing (which I think translates to an implicit `all`).